akka-stream/src/main/scala/akka/stream/javadsl/FlattenStrategy.scala

```diff
@@ -17,11 +17,11 @@ object FlattenStrategy {
    * emitting its elements directly to the output until it completes and then taking the next stream. This has the
    * consequence that if one of the input stream is infinite, no other streams after that will be consumed from.
    */
-  def concat[T, U]: FlattenStrategy[Source[T, U], T] = Concat[T, U]()
+  def concat[T, U]: FlattenStrategy[Source[T, U], T] = Concat.asInstanceOf[FlattenStrategy[Source[T, U], T]]
```

Comment:
@drewhk I think this change might be the only public api breaking change in this PR, but it is trivially fixable by keeping a private Concat instance and using that for `def concat`. Wydt?